compiler/internal/codegen/type_resolver_test.go

```diff
@@ -1,3 +1,5 @@
+//go:build go1.18
```

Comment:
Are some of the tests applicable to Go 1.17? Would be nice to run those in that case.